client/src/components/Sidebar/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { Sidebar } from './Sidebar';
+
+export function initSidebar() {
+  const element = document.getElementById('wagtail-sidebar');
+
+  const navigate = (url: string) => {
+    window.location.href = url;
+
+    // Return a promise that never resolves.
+    // This promise is used to indicate to any open submenus that the next page has loaded and it should close.
+    // As all navigation from the menu at the moment takes the user to another page, we don't need to close the menus.
+    // We will need to update this if we later add the ability to render views on the client side.
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    return new Promise<void>(() => {});
+  };
+
+  if (element instanceof HTMLElement && element.dataset.props) {
+    const props = (window as any).telepath.unpack(JSON.parse(element.dataset.props));
```

Comment:
Suggestion: You could extend the types with 
```ts
declare global {
    interface Window {
        telepath: any; // Ideally, we might want to add types for telepath someday :) 
    }
}
``` 
Here for example: https://github.com/wagtail/wagtail/blob/main/client/src/custom.d.ts